inch.go

```diff
@@ -0,0 +1,663 @@
+	msg := ""
+
+	for _, err := range el {
+		msg = fmt.Sprintf("%s%s\n", msg, err)
+	}
+
+	return msg
+}
+
+// Main represents the main program execution.
+type Inch struct {
+	mu             sync.Mutex
+	writtenN       int
+	startTime      time.Time
+	now            time.Time
+	timePerSeries  int64 // How much the client is backing off due to unacceptible response times.
+	currentDelay   time.Duration
+	wmaLatency     float64
+	latencyHistory []time.Duration
+	totalLatency   time.Duration
+	currentErrors  int   // The current number of errors since last reporting.
+	totalErrors    int64 // The total number of errors encountered.
+
+	Stdout io.Writer
+	Stderr io.Writer
+
+	// Client to be used to report statistics to an Influx instance.
+	clt client.Client
+
+	// Decay factor used when weighting average latency returned by server.
+	alpha float64
+
+	Verbose    bool
+	ReportHost string
+	ReportTags map[string]string
+	DryRun     bool
+	MaxErrors  int
+
+	Host             string
+	Consistency      string
+	Concurrency      int
+	Measurements     int   // Number of measurements
+	Tags             []int // tag cardinalities
+	PointsPerSeries  int
+	FieldsPerPoint   int
+	BatchSize        int
+	TargetMaxLatency time.Duration
+
+	Database      string
+	ShardDuration string        // Set a custom shard duration.
+	TimeSpan      time.Duration // The length of time to span writes over.
+	Delay         time.Duration // A delay inserted in between writes.
+}
+
+// NewMain returns a new instance of Main.
+func NewInch() *Inch {
+
+	// create an inch object with reasonable defaults
+	return &Inch{
+
+		alpha:          0.5, // Weight the mean latency by 50% history / 50% latest value.
+		latencyHistory: make([]time.Duration, 0, 200),
+
+		Consistency:     "any",
+		Concurrency:     1,
+		Measurements:    1,
+		Tags:            []int{10, 10, 10},
+		PointsPerSeries: 100,
+		FieldsPerPoint:  1,
+		BatchSize:       5000,
+		Database:        "db",
+		ShardDuration:   "7d",
+	}
+}
+
+// ParseFlags parses the command line flags.
+func (inch *Inch) Valid() error {
+	el := ErrorList{}
+
+	switch inch.Consistency {
+	case "any", "quorum", "one", "all":
+	default:
+		el = append(el, errors.New(`Consistency must be one of: {"any", "quorum", "one", "all"}`))
+	}
+
+	if inch.FieldsPerPoint < 1 {
+		el = append(el, errors.New("number of fields must be > 0"))
+	}
+
+	// validate reporting client is accessable
+	if inch.ReportHost != "" {
+		var err error
+		inch.clt, err = client.NewHTTPClient(client.HTTPConfig{
+			Addr: inch.ReportHost,
+		})
+		if err != nil {
+			el = append(el, fmt.Errorf("failed to communicate with %q: %s", inch.ReportHost, err))
+			return el
+		}
+
+		if _, err := inch.clt.Query(client.NewQuery(fmt.Sprintf(`CREATE DATABASE "ingest_benchmarks"`), "", "")); err != nil {
+			el = append(el, fmt.Errorf("unable to connect to %q: %s", inch.ReportHost, err))
+			return el
+		}
+	}
+
+	if len(el) > 0 {
+		return el
+	}
+
+	return nil
+}
```

Comment:
Can you rename this to `Validate()`? Also, it's probably worth calling from `Run()` instead of relying on the caller to invoke it.